lib/babel-istanbul-instrumenter.js

```diff
@@ -33,6 +32,13 @@ Instrumenter.prototype = Object.create(istanbul.Instrumenter.prototype);
 Instrumenter.prototype.constructor = Instrumenter;
 
 Instrumenter.prototype.instrumentSync = function(code, fileName) {
+  var plugins = this.babelOptions.plugins;
+  // If we're running in coverage, we're assuming that this is running in CI or in some
+  // form of test scenario and not being built for production. So it's fine to always
+  // force this plugin to work.
+  for (var plugin of this.plugins) {
```

Comment:
I think we are safe here, because this is using "for...of" rather than "for...in". "for...of" only iterates over the elements of the collection:

```javascript
Object.prototype.objCustom = function() {}; 
Array.prototype.arrCustom = function() {};

let iterable = [3, 5, 7];
iterable.foo = 'hello';

for (let i in iterable) {
  console.log(i); // logs 0, 1, 2, "foo", "arrCustom", "objCustom"
}

for (let i of iterable) {
  console.log(i); // logs 3, 5, 7
}
```